src/rtree.c

```diff
@@ -28,29 +28,29 @@ rtree_new(rtree_t *rtree, bool zeroed) {
 }
 
 static rtree_node_elm_t *
-rtree_node_alloc_impl(tsdn_t *tsdn, rtree_t *rtree, size_t nelms) {
+rtree_node_alloc_impl(tsdn_t *tsdn, UNUSED rtree_t *rtree, size_t nelms) {
 	return (rtree_node_elm_t *)base_alloc(tsdn, b0get(), nelms *
 	    sizeof(rtree_node_elm_t), CACHELINE);
 }
 rtree_node_alloc_t *JET_MUTABLE rtree_node_alloc = rtree_node_alloc_impl;
 
 static void
-rtree_node_dalloc_impl(tsdn_t *tsdn, rtree_t *rtree, rtree_node_elm_t *node) {
+rtree_node_dalloc_impl(UNUSED tsdn_t *tsdn, UNUSED rtree_t *rtree, UNUSED rtree_node_elm_t *node) {
```

Comment:
I'm not at all a fan of warnings about unused function parameters, and I'd rather we configure to disable such warnings when necessary, rather than bulking up our function declarations with mitigations.

Reply:
@jasone given that a lot of functions already do this, I think that disabling these globally might be a valid option. 

When updating these I see that many functions take parameters that they just never use. I can only guess why, but if the reasons for these are valid, disabling the warning globally for jemalloc is the way to go.

Reply:
In many cases it's indeed intentional, to keep similar functions have the same set of input.

Reply:
That's what it felt like. So maybe disabling this globally and removing all the `UNUSED` annotations is the way to go. It seems that a lot of code would require these annotations otherwise, and the wins are small: adding them did not find any bugs and it was painful to do so.

Reply:
So I've removed all usages of `UNUSED` function parameters. I've left unused for unused functions and unused variables, but those appear only 4 times in the code.

I've disabled the "unused function parameter" warning globally for each translation unit that includes the `jemalloc_internal_macros.h` header. It is never re-enabled since users cannot include this header anyways.